pkg/server/server.go

```
@@ -81,51 +156,110 @@ func (p PizzaOvenServer) handleRequest(w http.ResponseWriter, r *http.Request) {
+				http.Error(w, "Could not process input", http.StatusInternalServerError)
+				return
+			}
+		}()
+	}
+}
 
-		p.Logger.Debug("Could not validate repo URL %s: %s", data.URL, err.Error())
-		http.Error(w, fmt.Sprintf("not valid git repo URL. Expected format protocol://address but got: %s", err.Error()), http.StatusBadRequest)
+func (p PizzaOvenServer) handleBakeOrgRequest(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		p.Logger.Errorf("Received request with invalid method: %v", r.Body)
+		http.Error(w, "Invalid request method, expected post", http.StatusMethodNotAllowed)
 		return
 	}
 
+	var data orgReqData
+	err := json.NewDecoder(r.Body).Decode(&data)
+	if err != nil {
+		p.Logger.Errorf("Could not decode request json body: %v with error: %v", r.Body, err)
+		http.Error(w, "Could not decode request body", http.StatusBadRequest)
+		return
+	}
 	w.WriteHeader(http.StatusAccepted)
+	if data.Org == "" {
+		p.Logger.Error("Could not process blank org input")
+		http.Error(w, "Could not process input", http.StatusBadRequest)
+		return
+	}
+	cloneURLs, err := p.processOrg(data.Org, data.Archives)
+	if err != nil {
+		p.Logger.Errorf("Could not process org input: %v with error: %v", r.Body, err)
+		http.Error(w, "Could not process input", http.StatusInternalServerError)
+		return
+	}
 	if data.Wait {
-		err = p.processRepository(repoURLendpoint.String())
-		if err != nil {
-			p.Logger.Errorf("Could not process repository input: %v with error: %v", r.Body, err)
+		errorChannel := make(chan error)
+		wg := new(sync.WaitGroup)
+		wg.Add(len(cloneURLs))
+		for _, htmlURL := range cloneURLs {
+			go func(htmlURL string, wg *sync.WaitGroup, c chan error) {
+				_, repoURLendpoint, err := p.getValidRepoEndpoint(htmlURL)
+				if err != nil {
+					c <- err
+				} else {
+					c <- p.processRepository(repoURLendpoint)
+				}
+				defer wg.Done()
+			}(htmlURL, wg, errorChannel)
+		}
+		defer wg.Wait()
```

You shouldn't `defer` here: this won't block on waiting for all the go routines to finish in the waitgroup and will continue on without finishing processing squashing any potential errors that might arrise in the `errorChannel`